### ISSUE TITLE
C# Collection type as any[]

### ISSUE THREAD

**Issue by maleet** [frhagn/Typewriter#131]
I got an old project with lots of custom collections like:
```
public class MessagePackEntity
    {
        private MessageEntityCollection infoMessages;
        public MessageEntityCollection InfoMessages
        {
            get
            {
                if (infoMessages == null)
                {
                    infoMessages = new MessageEntityCollection();
                }

                return infoMessages;
            }
            set
            {
                infoMessages = value;
            }
        }
        ....
}

public class MessageEntityCollection : List<MessageEntity>
{
 ....
}
```

result on rendering:

```
export class MessagePackEntity {
    
    // InfoMessages
    public infoMessages: any[];
    ...
}
```

in tst file
```
$Classes([TypeWriter])[
export class $Name {
    $Properties[
    // $Name
    public $name: $Type;]
}]
```

Would like to have 
```
export class MessagePackEntity {
    
    // InfoMessages
    public infoMessages: messageEntity[];
    ...
}
```
Is it possible?

**Comment by aluanhaddad** [comment]
Yes it is. If this is an external module, you need to generate a model type for the collection element and import it.
This works well for me:


_helpers_
```C#
IEnumerable<Type> PropertyTypes(Class @class) =>
        @class.Properties
            .Select(x => x.Type)
            .Select(x => x.Unwrap())
            .Where(x => !x.IsPrimitive || x.IsDate || x.IsEnum)
            .GroupBy(x => x.Name).Select(x => x.First());

    IEnumerable<Type> PropertyTypes(Interface @interface) =>
        @interface.Properties
            .Select(t => t.Type)
            .Select(t => t.Unwrap())
            .Where(x => !x.IsPrimitive || x.IsDate || x.IsEnum)
            .GroupBy(x => x.Name).Select(x => x.First());

string PropertyTypeName(Property property) => 
    property.Type.Name == "Date" ? "Moment | string" : property.Type.Name;

string ImportTypeName(Type type) => type.Name == "Date" ? "{ Moment }" : type.Name;

string FileName(Class @class) => "./" + FileNameFromTypeName(@class.name);
string FileName(Interface @interface) => "./" + FileNameFromTypeName(@interface.name);
string FileName(Property property) => "./" + FileNameFromTypeName(property.Type.name);
string FileName(Type type) => type.IsDate? "moment" :  "./" + FileNameFromTypeName(type.name);
string FileName(string type) => "./" + FileNameFromTypeName(type);

```

_template_
```C#
$Classes[
$PropertyTypes[
import $ImportTypeName from '$FileName';
]

export interface $Name {
    $Properties[$name?: $PropertyTypeName;
    ]
}
export default $Name;
]
```

**Comment by frhagn** [comment]
Resolved in 1.8.0
